libr/anal/var.c

```diff
@@ -370,7 +370,7 @@ R_API RList *r_anal_var_list(RAnal *a, RAnalFunction *fcn, int kind) {
 	char *varlist;
 	RList *list = r_list_new ();
 	if (!a || !fcn)
-		return list;
+		return NULL;
```

Comment:
missing r_list_free.. or well. we should construct `list = r_list_new` after this check. not before